usaspending_api/common/query_with_filters.py

```diff
@@ -120,6 +165,26 @@ def generate_elasticsearch_query(cls, filter_values: List[str], query_type: _Que
         return ES_Q("dis_max", queries=keyword_queries)
 
 
+class _TransactionKeywordSearch(_Filter):
```

Comment:
Doesn't seem like this is compatible with a `_QueryType` of `transactions`. I think we should either:
- Make this compatible with Transactions
- Make it clear that it's not compatible with Transactions and potentially throw an error if this is attempted like you did with `recipient_id` for `subaward`

Also, I don't think this filter is functional yet because it is being filtered out during request validations. I think adding it to `AWARD_FILTER` in `usaspending-api/usaspending_api/common/validator/award_filter.py` may help with this. Be careful though because Transaction based endpoints use the same object to validate their requests. As far as I know, before this PR all filters in this file were compatible with all `_QueryType`s, with the exception of `subaward` and `award` not supporting `recipient_id`.

Reply:
The name of the class indicates what it does. It's not the right interpretation to think that because Transaction is in the class name that it's referring to QueryTypes.

This class name is meant to show that a keyword search is being done on transactions.

Reply:
I don't like either of those options.

Reply:
>The name of the class hints at what it does, that was my goal at least and why I renamed it from it's old, existing name. It's not the right interpretation to think of the name as meaning Transaction from QueryTypes.
This class name is meant to show that a keyword search is being done on transactions.

I think the rename was a good move and works. My point here is that it's breaking the pattern of the other filters to support all `_QueryTypes` without any indication of doing so. In the `recipient_id` filter, you left a comment and threw an exception when trying to use an incompatible `_QueryType`, which I thought was really useful for readers of the code. 
```
if query_type == _QueryType.SUBAWARDS:
    # Subawards did not support "recipient_id" before migrating to elastic search
    # so this behavior is honored here.
    raise InvalidParameterException(
        f"Invalid filter: {_RecipientId.underscore_name} is not supported for subaward queries."
    )
```

Reply:
> What you see is what existed before this story. If it is not supported now it means it never was. Adding support for it is not within the scope of this ticket. FYI this filter is some sort of internal concept, our code made me believe it's not actually a filter we should have in our api contracts.

I was just calling this out because you already did most of the work implementing this filter: renaming it in contracts and some usages across the codebase, implementing the filter function. I thought you wanted to see it go live and might have just missed this one piece of it.

I'm okay if you don't want to make this filter functional. Can you please just remove it from the API contracts then and add a ticket to the tech-debt backlog to decide whether we can remove the filter altogether or not?

Reply:
It was already in API contracts. I shouldn't remove it. Doing so would require FE approval and collaboration, right? Is this more scope creep?

No, it's not right for me to make this filter functional because it scope creeps my story. I'm happy to work on an additional story that has AC for it to be functional

Reply:
I see your point of view about the API contracts. 

The only thing that was not addressed was my feedback around making it clear that `_TransactionKeywordSearch` is not compatible with the transaction `_QueryType`. I'm okay if you don't want to enforce this in the code, but please leave a comment on that filter calling out what this filter does (which is not immediately obvious) and that it doesn't support the transaction `_QueryType`.

Reply:
This is compatible with the award query type as well.